src/xtgeo/cube/_cube_import.py

```diff
@@ -20,103 +48,105 @@ def import_segy(sfile: xtgeo._XTGeoFile) -> Dict:
     """Import SEGY via the SegyIO library.
 
     Args:
-        sfile (str): File name of SEGY file
+        sfile: File object for SEGY file
     """
-    # pylint: disable=too-many-statements
-    # pylint: disable=too-many-locals
     sfile = sfile.file
 
-    logger.debug("Inline sorting %s", segyio.TraceSortingFormat.INLINE_SORTING)
+    attributes = dict()
+
+    try:
+        # cube with all traces present
+        with segyio.open(sfile, "r") as segyfile:
+            attributes = _import_segy_all_traces(segyfile)
+    except ValueError as verr:
+        if any([word in str(verr) for word in ["Invalid dimensions", "inconsistent"]]):
+            # cube with missing traces
+            print(
+                "XTGeo notice: Missing or inconsistent traces in SEGY, will try "
+                "but check result carefully!"
+            )
 
-    with segyio.open(sfile, "r") as segyfile:
-        segyfile.mmap()
+            with segyio.open(sfile, "r", ignore_geometry=True) as segyfile:
+                attributes = _import_segy_incomplete_traces(segyfile)
+        else:
+            raise
+    except BaseException as berr:
```

Comment:
Is catching `BaseException` called for here? Is always better to catch spesific exceptions, and `BaseException` includes `KeyboardInterupt` and `SystemExit`, so is usually never used outside the entry point. If you must catch all, `Exception` is usually the way to go.